### PR TITLE
Migrate away from the deprecated `django-heroku` package

### PR DESCRIPTION
**Pull request by PrateekSane**
The `django-heroku` package is deprecated/sunset since it's no longer the preferred way for configuring Django on Heroku. As such, the config from `django-heroku` has now been inlined into this project and simplified.

GUS-W-8012597.

### CODE REVIEW

File: requirements.txt
Comment (edmorley) [inline requirements.txt:6]: I don't think `pytest` was installed before - unless we need it here for some reason, I'd lean towards omitting it? 

Also, whilst django-heroku didn't do this, I think we might want to start adding version ranges to all of the packages, so they don't suddenly start breaking if a new major version is released. We can always also enable Dependabot for this repo (I'll file a new GUS work item for that shortly).

```suggestion
dj-database-url>=0.5,<0.6
whitenoise>=6.0,<7.0
psycopg2>=2.0,<3.0
```

File: requirements.txt
Comment (edmorley) [inline requirements.txt:2]:
```suggestion
django>=4.0,<5.0
gunicorn>=20.0,<21.0
```

File: django_heroku.py
Comment (edmorley) [inline django_heroku.py:44]: I'm not convinced we need this at all.

Heroku CI DBs are ephemeral, so shouldn't need any cleanup.

Could you see what happens for the Heroku CI run if this is just removed entirely? (plus the corresponding `TEST_RUNNER` setting part below removed too)

File: django_heroku.py
Comment (edmorley) [inline django_heroku.py:47]: Since django-heroku is now inlined, I don't think we need these configuration options any more (they make more sense when it's a library - but for the config here, users will just edit the config inline in their own git clone of this repo, so no need for toggleable options etc).

Could you remove `settings()` and inline everything in the file (along with removing any redundant conditionals)? I would keep the defaults as they are. ie: If a feature was enabled by default, then we keep it, otherwise we delete the feature (so the `db_colors` feature would be deleted).

File: django_heroku.py
Comment (edmorley) [inline django_heroku.py:67]: Let's delete this feature - most users won't be using multiple DBs when starting out

File: django_heroku.py
Comment (edmorley) [inline django_heroku.py:87]: Lets try removing this/the test runner class above and seeing how Heroku CI reacts.

File: django_heroku.py
Comment (edmorley) [inline django_heroku.py:106]: I would rewrite the whitenoise config to match the best practices in the upstream whitenoise docs:
https://whitenoise.evans.io/en/stable/#quickstart-for-django-apps

**Review by edmorley — CHANGES_REQUESTED** [review]
Thank you for working on this! :-)

File: requirements.txt
Comment (PrateekSane) [reply requirements.txt]: made change. tested by create venv and installing to make sure everything installs properly. will make change to the guide draft as well

File: requirements.txt
Comment (PrateekSane) [reply requirements.txt]: same as previous

File: django_heroku.py
Comment (PrateekSane) [reply django_heroku.py]: added comment and the if statement

File: django_heroku.py
Comment (PrateekSane) [reply django_heroku.py]: Weird error. Will discuss tmr

File: django_heroku.py
Comment (PrateekSane) [reply django_heroku.py:64]: same as other. will discuss tmr

File: django_heroku.py
Comment (PrateekSane) [reply django_heroku.py]: Kept the class of HerokuDiscoverRunner but cut out functions _wipe_tables and teardown_databases. When trying to remove the entire functions, if self.keepdb is not kept as true the following error pops up. This has also been moved to settings.py
`-----> Running test command `python manage.py test`...
Found 1 test(s).
Creating test database for alias 'default'...
/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/postgresql/base.py:323: RuntimeWarning: Normally Django will use a connection to the 'postgres' database to avoid running initialization queries against the production database when it's not needed (for example, when running tests). Django was unable to create a connection to the 'postgres' database and will use the first PostgreSQL database instead.
  warnings.warn(
Got an error creating the test database: permission denied to create database
-----> test command `python manage.py test` failed with exit status 2`

File: gettingstarted/settings.py
Comment (edmorley) [inline gettingstarted/settings.py:62]: The order of `MIDDLEWARE` affects the order in which it runs, which can make a big difference.

The middleware is now being inserted at the end, whereas django-heroku inserted the WhiteNoise middleware at the beginning:
https://github.com/heroku/django-heroku/blob/fa2a5f7719625020a0dcf29654253f2982c26557/django_heroku/core.py#L105

...and in fact even at the beginning wasn't quite right. I would see the WhiteNoise docs for best practices for how to configure WhiteNoise - mentioned in the last review:
https://github.com/heroku/python-getting-started/pull/167#discussion_r925761263

After doing that, I would also add a comment here explaining what WhiteNoise does (given people new to Django or Heroku might not know, and it's something not in the template project produced by `django-admin startproject` etc):
```
# WhiteNoise handles serving static files in production.
```

File: gettingstarted/settings.py
Comment (edmorley) [inline gettingstarted/settings.py:136]: With Python 3's `Path`, this should now work (and is what the WhiteNoise docs say to do):
https://whitenoise.evans.io/en/stable/django.html

```suggestion
STATIC_ROOT = BASE_DIR / "staticfiles"
```

File: gettingstarted/settings.py
Comment (edmorley) [inline gettingstarted/settings.py:137]: (To match what `django-admin startproject` generates) 

```suggestion
STATIC_URL = "static/"
```

File: gettingstarted/settings.py
Comment (edmorley) [inline gettingstarted/settings.py:142]:
```suggestion
```

This isn't needed since `collectstatic` automatically makes the directory. This can be confirmed by making sure the directory doesn't exist locally, then running `./manage.py collectstatic` and seeing it succeeds,  eg:

```
(.venv) gettingstarted $ ./manage.py collectstatic

128 static files copied to '/Users/emorley/src/django-test/gettingstarted/staticfiles', 378 post-processed.
```

File: gettingstarted/settings.py
Comment (edmorley) [inline gettingstarted/settings.py:144]:
```suggestion
# Enable WhiteNoise's GZip compression of static assets.
```

**Review by edmorley — CHANGES_REQUESTED** [review]
The updates look great - almost there! :-)

File: gettingstarted/settings.py
Comment (PrateekSane) [reply gettingstarted/settings.py:30]: If it is in heroku is there always a secret key?

File: gettingstarted/settings.py
Comment (PrateekSane) [reply gettingstarted/settings.py]: moved to under securitymiddleware like the docs

File: gettingstarted/settings.py
Comment (PrateekSane) [reply gettingstarted/settings.py]: As per Ed and I discussed we will keep the hardcoded due to git push issues

File: gettingstarted/settings.py
Comment (edmorley) [inline gettingstarted/settings.py]:
```suggestion
Generated by 'django-admin startproject' using Django 4.0.
```

File: gettingstarted/settings.py
Comment (edmorley) [inline gettingstarted/settings.py]:
```suggestion
# https://docs.djangoproject.com/en/4.0/ref/settings/#databases
```

File: gettingstarted/settings.py
Comment (edmorley) [inline gettingstarted/settings.py]:
```suggestion
# https://docs.djangoproject.com/en/4.0/ref/settings/#auth-password-validators
```

**Review by edmorley — APPROVED** [review]
Thank you for doing this!

I've applied the last few review comments now, just so we can merge this straight away without needing to wait for another review cycle.

File: gettingstarted/settings.py
Comment (edmorley) [reply gettingstarted/settings.py]: There were still some links that needed updating, so I fixed these in d562bca.

**Comment by edmorley** [comment]
I've promoted your corresponding staged Dev Center changes to the main article:
https://devcenter.heroku.com/articles/getting-started-with-python
https://devcenter.heroku.com/admin/articles/diff/2161/2161v165..2161v166

**Pull request merged**
